### PR TITLE
feat(filters): dynamic filter components - part 2

### PR DESCRIPTION
**Pull request by Birkbjo**

### CODE REVIEW

File: src/components/sectionList/filters/filterSelectors/PublicAccessFilter.tsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import i18n from '@dhis2/d2-i18n'
+import React from 'react'
+import { ConstantSelectionFilter } from './ConstantSelectionFilter'
+
+const constants = {
+    'rw------': 'Public can edit',
+    'r-------': 'Public can view',
+    '--------': 'Public cannot access',
+}
```
Comment (Birkbjo) [inline src/components/sectionList/filters/filterSelectors/PublicAccessFilter.tsx:9]: Note that this currently worldn't work for objects that also have "data sharing" (like category options). This was just a simple implementation for now.

**Review by Mohammer5 — COMMENTED** [review]
I just gave it a try. I added the "public access" filter and removed the "domain type" filter. But the domain type filter is still showing.

**Comment by Birkbjo** [comment]
> I just gave it a try. I added the "public access" filter and removed the "domain type" filter. But the domain type filter is still showing.

I just did this on the preview branch and it works. Will will update and fix conflicts in this branch though.

Did you select a domain type value? If a filter has a value, it will always show.

**Comment by Mohammer5** [comment]
Yeah, now it works for me as well. Not sure what happened, maybe I did select a value.
I suppose we could disable selected options in the transfer that have a value, but that's a question for @cooper-joe. For now it's fine I guess

**Review by Mohammer5 — APPROVED** [review]

File: src/lib/dataStore/useDataStore.ts [hunk 2]
```diff
@@ -102,6 +102,7 @@ export function useDataStoreValuesQuery<
         queryKey: [query],
         queryFn: createBoundQueryFn(engine),
         placeholderData,
+        // keepPreviousData: true,
```
Comment (Birkbjo) [inline src/lib/dataStore/useDataStore.ts:105]: Todo: fix

**Pull request merged**